### PR TITLE
Renamed requestableResources to resources in clusterQueue

### PR DESCRIPTION
**Pull request by ahg-g**
#### What type of PR is this?

/kind cleanup
/kind api-change

#### What this PR does / why we need it:

Renames clusterQueue.spec.requestableResources to just resources. This is simpler and the docs should be enough to clarify that the limits are defined against the resources requested by workloads.

#### Which issue(s) this PR fixes:
Fixes #

### CODE REVIEW

File: docs/concepts/cluster_queue.md [hunk 1]
```diff
@@ -103,10 +103,10 @@ taints:
 ```
 
 You can use the `.metadata.name` to reference a flavor from a ClusterQueue in
-the `.spec.requestableResources[*].flavors[*].name` field.
+the `.spec.resources[*].flavors[*].name` field.
 
 For each resource of each [pod set](workload.md#pod-sets) in a Workload, Kueue
-assigns the first flavor in the `.spec.requestableResources[*].resources.flavors`
+assigns the first flavor in the `.spec.resources[*].resources.flavors`
```
Comment (alculquicondor) [inline docs/concepts/cluster_queue.md:109]: my bad

```suggestion
assigns the first flavor in the `.spec.resources[*].flavors`
```

**Review by alculquicondor — COMMENTED** [review]
Missing a change in `config/samples`

File: docs/concepts/cluster_queue.md (hunk 1 above)
Comment (ahg-g) [reply docs/concepts/cluster_queue.md]: done.

**Comment by ahg-g** [comment]
> Missing a change in `config/samples`

updated

**Review by alculquicondor — COMMENTED** [review]
/lgtm

**Pull request merged**